### PR TITLE
sw4lite: fix build errors and add variants

### PR DESCRIPTION
**Pull request by bhatele**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sw4lite/package.py [hunk 1]
```diff
@@ -23,32 +23,76 @@
...
+    variant('precision',
+        default='double',
+        values=('float', 'double'),
+        multi=False,
+        description='Floating point precision')
+    variant('ckernel',
+        default='no',
+        values=('yes', 'no'),
+        multi=False,
+        description='C or Fortran kernel')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/sw4lite/package.py]: Just use the default boolean variant syntax, with default `False` and no values specified.

File: var/spack/repos/builtin/packages/sw4lite/package.py [hunk 2]
```diff
@@ -23,32 +23,76 @@
...
         targets.append('FC=' + spec['mpi'].mpifc)
         targets.append('CXX=' + spec['mpi'].mpicxx)
+
+        targets.append('CFLAGS={0}'.format(' '.join(cflags)))
+        targets.append('CXXFLAGS={0}'.format(' '.join(cxxflags)))
+        targets.append('FFLAGS={0}'.format(' '.join(fflags)))
+
+        targets.append('EXTRA_CXX_FLAGS=')
+        targets.append('EXTRA_FORT_FLAGS=')
+        targets.append('EXTRA_LINK_FLAGS=-lblas -llapack -lifcore')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sw4lite/package.py]: you probably change by accident the same file in two different PRs, either way, same as  https://github.com/LLNL/spack/pull/5945#discussion_r146760288

File: var/spack/repos/builtin/packages/sw4lite/package.py [hunk 3]
```diff
@@ -23,32 +23,75 @@
...
         targets.append('FC=' + spec['mpi'].mpifc)
         targets.append('CXX=' + spec['mpi'].mpicxx)
+
+        targets.append('CFLAGS={0}'.format(' '.join(cflags)))
+        targets.append('CXXFLAGS={0}'.format(' '.join(cxxflags)))
+        targets.append('FFLAGS={0}'.format(' '.join(fflags)))
+
+        targets.append('EXTRA_CXX_FLAGS=')
+        targets.append('EXTRA_FORT_FLAGS=')
+        blas_lapack = spec['blas'].libs + spec['lapack'].libs
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sw4lite/package.py]: should be other way around, `lapack` depends on `blas`. The order of addition here is important.

**Comment by tgamblin** [comment]
LGTM.

Minor comment: variants look better when not wrapped so much.  Also there is no length restriction on variant lines, so you don't have to wrap them.

Also, `flake8` checks are failing.

**Pull request merged**